### DISCUSSION TITLE
Integrated Hasura

### DISCUSSION THREAD

**Discussion by jtkeyva** [supabase/supabase#7256] [Feature Requests]
I used [Hasura](https://hasura.io) before I discovered Supabase and it's awesome. I've even [connected Hasura to Supabase](https://www.youtube.com/watch?v=FKM4qVJFSAs) and really enjoy the UI amongst connecting multiple databases, realtime subscriptions and more.

Seems like Hasura integration would be a perfect fit for Supabase. Any chances of that happening?

Thanks

**Comment by w3b6x9** [comment]
I don't believe there's been discussions around integrating with Hasura. Supabase recently launched its GraphQL support. See https://supabase.com/blog/2022/03/29/graphql-now-available.

**Reply by jtkeyva** [reply]
Yeah, I've been starting to play with it. GraphiQL is a must :)

**Reply by dthyresson** [reply]
> GraphiQL is a must :)

@jtkeyva When GraphQL support launched last launch week, I worked on a demo Hackernews-clone app and as part of that integrated GraphiQL as part of the project; see: https://github.com/supabase-community/supabase-graphql-example#graphiql

That said, I can see that having GraphiQL as part of the Supabase studio (local or hosted) would be helpful.

**Reply by jtkeyva** [reply]
cool 😎

**Reply by ilbertt** [reply]
I really appreciate the **pg_graphql** extension launched by Supabase, but I think **Hasura** still has more features that are really useful, [realtime GraphQL subscriptions](https://hasura.io/docs/latest/graphql/core/databases/postgres/subscriptions/index/) for example. As I also explained in a [Discord thread](https://discord.com/channels/839993398554656828/962087531363565618/962294511868993577), we know that Supabase offers the possibility to [subscribe to realtime changes in the database](https://supabase.com/docs/reference/javascript/subscribe), but they should be exposed also to the GrpahQL endpoint in order to have a single source of truth for the API in the frontend and also leverage on Typescript types generation with tools like [GraphQL Code Generator](https://www.graphql-code-generator.com/).
That's why I'd still be interested in integrating Hasura with Supabase

**Reply by jtkeyva** [reply]
Ya, seems like Hasura integration could leap frog a lot of dev time and resources... and it's open source

**Reply by FloNeu** [reply]
I think this would be cool - as i am also using hasura and just stumbled over supabase - when looking for an easy way to extend it with an auth and file-upload service... maybe it would dublicate some stuff already provided by supabase/hasura - but the graphql features of hasura a probably more advanced then what is available with supabase graphql support - like query-caching, allowing max query depth/value, restricting to known queries and more.

**Reply by zlanich** [reply]
I too am very interested in first party integration of Hasura. I can see that there's a lot of overlap with auth, events, realtime, etc, so Idk the idea will gain traction, but after using Hasura for a while in production, my hesitation with using Supabase lies in moving away from all the niceties of Hasura toward other less feature rich open source alternatives like pg_graphql. The row-level security, remote schemas, live queries, etc in Hasura are REALLY nice. They are also coming out with a V3 that might provide more leaps. I'd be interested to hear people's thoughts.

**Comment by governoze** [comment]
what I really like on Hasura that make a perfect fit for supabase is the data federation engine
